### PR TITLE
Adding Extensions to Support Main Pipeline

### PR DESCRIPTION
**Pull request by Sohambutala**
These extensions are designed to trigger or support the main pipeline, aiding in data source acquisition, enabling concurrent deployments etc.

**1. File Monitoring for Automated Deployments:**

  **Purpose**: To monitor file changes and trigger specific deployments.
  
  **Current Approach**:
  - **Schedule**: watchdog is not used currently as we only need to schedule at regular intervals. This can be changed later, or a separate extension can be added for real-time scenarios.
  - **Deployment Independence**: Each file triggers a new deployment, and a failure in one deployment does not affect others.
  
  **Implementation Details**:
  
  - Tracking: Utilizes `Prefect Blocks` to record the last successful run time and the list of processed files.
  - Concurrency: Leverages task concurrency to run multiple deployments simultaneously, allowing for concurrent file processing.
  - Fail-Safe Option: The `fail_safe` option toggles between concurrent and sequential processing. In sequential mode, a failure in one file stops the processing of remaining files. In concurrent mode, file processing is independent.
  
  **Considerations**:
 Concurrency must be managed carefully when appending results to a single Zarr store to avoid race conditions and maintain the order of data.
  

**2. File Downloader for Seamless Data Transfer:**

**Purpose**: To facilitate data transfer between two locations.

**Implementation Details**:
  
  - Tracking: Checks the destination folder for existing files and skips the transfer if the file is already present.
  - Concurrency: Employs task concurrency to download multiple files simultaneously. Concurrency can be managed by configuring the global concurrency limit (`edf-data-transfer`).
  - Delete Option: Offers an option to delete files after transfer, with a default setting of `false`.
  
 **Considerations**:
  Be mindful of rate limitations of both the source and destination. Concurrent requests can trigger rate limits.

### CODE REVIEW

**Comment by Sohambutala** [comment]
Converting this PR to draft since Prefect variables have a limit of 255 characters. An alternative would be to try using the Echodataflow block object.

File: echodataflow/models/output_model.py [hunk 1]
```diff
@@ -1,90 +1,29 @@
...
 Classes:
-    ErrorObject (BaseModel): Model for representing an error object.
-    EchodataflowObject (BaseModel): Model for representing an echodataflow object.
-    Group (BaseModel): Model for representing a group.
     Output (BaseModel): Model for representing output data.
 
 Author: Soham Butala
 Email: sbutala@uw.edu
-Date: June 5, 2024
+Date: August 22, 2023
```
Comment (valentina-s) [inline echodataflow/models/output_model.py:11]: older version: is the merge correct?

**Comment by valentina-s** [comment]
This PR has some files that seems from an older version. I point to one example in the comment above, but I also notice some of the actions versions got reversed to the older ones. So maybe you check this, before merging.

File: echodataflow/models/output_model.py (hunk 1 above)
Comment (Sohambutala) [reply echodataflow/models/output_model.py:11]: This particular file was originally dated August 22, I made a lot of change to the same file. There are so some stages with this date and its due to the code generation from CLI which has a hardcoded date. I will make a new issue for overall upgrade of the code generation CLI command.

**Comment by Sohambutala** [comment]
Thank you for highlighting workflow changes, I had those locally but never committed.

**Pull request merged**